### PR TITLE
Fixes docker events since beginning of unix time

### PR DESCRIPTION
**Pull request by aanm**
Fixes issue #11555
Applied a workaround to check if since and until flags are valid or not.

Signed-off-by: André Martins martins@noironetworks.com

### CODE REVIEW

**Comment by duglin** [comment]
Looks good but can we get a testcase for this?

**Comment by aanm** [comment]
@duglin 
One of my tests from https://github.com/docker/docker/pull/10511 is failing because of this bug. So I think the testcase is already done on a different branch. What do you think?

**Comment by duglin** [comment]
@aanm ok guess we can fake it until #10511 is in.  :-)
LGTM

**Comment by duglin** [comment]
ping @jfrazelle

File: events/events.go [hunk 1]
```diff
@@ -47,9 +47,11 @@ func (e *Events) Install(eng *engine.Engine) error {
 
 func (e *Events) Get(job *engine.Job) engine.Status {
 	var (
-		since   = job.GetenvInt64("since")
-		until   = job.GetenvInt64("until")
-		timeout = time.NewTimer(time.Unix(until, 0).Sub(time.Now()))
+		since       = job.GetenvInt64("since")
+		since_valid = job.Getenv("since") != ""
```
Comment (ahmetalpbalkan) [inline events/events.go]: these should be sinceValid and untilValid per Go naming conventions

Comment (aanm) [inline events/events.go]: Thanks! That's what I get for switching between python and docker :-)

File: events/events.go [hunk 2]
```diff
@@ -47,9 +47,11 @@ func (e *Events) Install(eng *engine.Engine) error {
 
 func (e *Events) Get(job *engine.Job) engine.Status {
 	var (
-		since   = job.GetenvInt64("since")
-		until   = job.GetenvInt64("until")
-		timeout = time.NewTimer(time.Unix(until, 0).Sub(time.Now()))
+		since      = job.GetenvInt64("since")
+		sinceValid = job.Getenv("since") != ""
```
Comment (LK4D4) [inline events/events.go]: Actually I'm not fan of `var`. Can we just use if `job.Getenv("since") != ""` below?

**Comment by aanm** [comment]
@duglin  
Since https://github.com/docker/docker/pull/10511 was closed, I will move the test here

File: events/events.go (hunk 2 above)
Comment (aanm) [inline events/events.go]: @LK4D4 
Sure, same with "untilValid", right?

**Comment by icecrime** [comment]
LGTM

**Pull request merged**